package.json

```diff
@@ -1,67 +1,79 @@
 {
-  "name": "mini-html-webpack-plugin",
-  "version": "0.0.0-development",
-  "description": "A miniature version of html-webpack-plugin with only necessary features",
-  "main": "index.js",
-  "files": [
-    "index.js"
-  ],
-  "scripts": {
-    "pretest": "npm run lint",
-    "test": "true && npm run test:jest",
-    "posttest": "npm run format",
-    "lint": "eslint . --cache --fix",
-    "format": "prettier --write \"**/*.{js,md}\"",
-    "test:jest": "jest",
-    "test:watch": "jest --watch",
-    "test:coverage": "jest --coverage"
-  },
-  "keywords": [
-    "webpack",
-    "html",
-    "template"
-  ],
-  "author": "Juho Vepsalainen",
-  "contributors": [
-    {
-      "name": "Artem Sapegin",
-      "url": "http://sapegin.me"
-    }
-  ],
-  "license": "MIT",
-  "dependencies": {
-    "webpack-sources": "^1.3.0"
-  },
-  "homepage": "https://github.com/styleguidist/mini-html-webpack-plugin",
-  "repository": "styleguidist/mini-html-webpack-plugin",
-  "engines": {
-    "node": ">=8.12"
-  },
-  "devDependencies": {
-    "@webpack-contrib/test-utils": "^0.1.3",
-    "css-loader": "^3.4.2",
-    "eslint": "^5.16.0",
-    "eslint-config-tamia": "^6.0.2",
-    "husky": "^2.3.0",
-    "jest": "^24.8.0",
-    "lint-staged": "^8.1.7",
-    "mini-css-extract-plugin": "^0.9.0",
-    "prettier": "^1.17.1",
-    "webpack": "^4.31.0"
-  },
-  "lint-staged": {
-    "*.{js,md}": [
-      "prettier --write",
-      "git add"
-    ],
-    "*.js": [
-      "eslint --fix",
-      "git add"
-    ]
-  },
-  "husky": {
-    "hooks": {
-      "pre-commit": "lint-staged"
-    }
-  }
+	"name": "mini-html-webpack-plugin",
+	"version": "0.0.0-development",
+	"description": "A miniature version of html-webpack-plugin with only necessary features",
+	"main": "dist/index.js",
+	"types": "dist/index.d.ts",
+	"files": [
+		"dist/index.js",
+		"dist/index.d.ts"
+	],
+	"scripts": {
+		"build": "rimraf ./dist && tsc",
+		"watch": "tsc --watch",
+		"pretest": "npm run lint",
+		"test": "true && npm run test:jest",
+		"posttest": "npm run format",
+		"lint": "eslint **/*.ts --cache --fix",
+		"format": "prettier --write \"**/*.{ts,md}\"",
+		"test:jest": "jest --detectOpenHandles",
```

Comment:
`--detectOpenHandles` is a weird one. Even with one test, Jest gives:

```
Jest did not exit one second after the test run has completed.

This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.
```

I am not sure where it's coming from, though.

Reply:
Maybe we're not waiting for a Promise somewhere?

Reply:
I'm not sure yet. I'll try against the `master` version if it's there too. It could be `@webpack-contrib/test-utils`.

I'll dig further as there are more moving bits now.

Reply:
Yeah, it's the compiler:

```js
test('default options', async () => {
	await compiler({}, {});

	expect(true).toEqual(true);
});
```

Weird.

Reply:
Looks like `master` doesn't have it.

Reply:
I know `@webpack-contrib/test-utils` has been deprecated by now so maybe one option would be to rewrite that one out.

Reply:
I reduced it to:

```
import compiler from '@webpack-contrib/test-utils';

test.only('default options', () => {
	return compiler({}, {}).then((result) => {
		console.log(Object.keys(result.compilation.assets));
	});
});
```

and the warning still appears without `--detectOpenHandles`.

I made sure all dependencies are up to date so that's not it.

Reply:
Weird ;-/

Reply:
I solved it. All I had to do was to remove `testEnvironment: 'node',` from Jest configuration as that was messing it up.

Reply:
I think what was happening is that the version of Node I have was getting confused with async/await. I imagine now it's compiling it to promises so it works as we expect.

Reply:
Why do we need to compile it to promises though? All versions of Node we care about support async/await.

Reply:
I am not sure. It can be a bug somewhere in Jest. We didn't set `testEnvironment` before and tests run fine (although against jsdom as far as I understand) so I would likely leave it like this.